src/mantle/framework/events/class-discover-events.php

```diff
@@ -80,23 +78,22 @@ protected static function get_listener_events( $listeners, string $base_path ):
 			}
 
 			foreach ( $listener->getMethods( ReflectionMethod::IS_PUBLIC ) as $method ) {
-				// Check for attribute support with PHP 8.
-				if ( version_compare( phpversion(), '8.0.0', '>=' ) ) {
-					// Check if the method has an attribute action.
-					$action_attributes = $method->getAttributes( Action::class );
-
-					if ( ! empty( $action_attributes ) ) {
-						foreach ( $action_attributes as $attribute ) {
-							$instance = $attribute->newInstance();
-
-							$listener_events[ $listener->name . '@' . $method->name ] = [
-								[ $instance->action ],
-								$instance->priority,
-							];
-						}
-
-						continue;
+				// Check if the method has an attribute action.
+				$action_attributes = $method->getAttributes( Action::class );
+
+				if ( ! empty( $action_attributes ) ) {
```

Comment:
I assume we know that the `getAttributes()` method will always return an iterable so we don't have to check the type before running `foreach` on it. But if that's not safe to assume, recommend checking the type here.